search/collector/topn_test.go

```diff
@@ -724,6 +727,48 @@ func TestCollectorChaining(t *testing.T) {
 	}
 }
 
+func setupIndex(t *testing.T) index.Index {
+	analysisQueue := index.NewAnalysisQueue(1)
+	i, err := scorch.NewScorch(
+		gtreap.Name,
+		map[string]interface{}{
+			"path": "",
+		},
+		analysisQueue)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = i.Open()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return i
+}
+
+func TestSetFacetsBuilder(t *testing.T) {
```

Comment:
Good test case, would you add an full index and search test case in search_test.go as well ..

- create an index
- add a few documents with doc values etc.
- search request with sort and facets
- make sure facet counts are accurate